src/checkbox/styled-components.js

```diff
@@ -226,12 +226,20 @@ export const Toggle = styled('div', props => ({
 }));
 
 export const ToggleInner = styled('div', props => {
-  function backgroundColor($theme) {
-    if (props.$disabled) return props.$theme.colors.mono400;
-    if (props.$isActive) return props.$theme.colors.primary500;
-    if (props.$isHovered || props.$checked) return props.$theme.colors.primary;
+  function backgroundColor() {
+    if (props.$disabled) {
+      return props.$theme.colors.sliderTrackFillDisabled;
+    }
+
+    if (props.$isActive && props.$checked) {
+      return props.$theme.colors.sliderTrackFillSelectedActive;
+    }
+
+    if (props.$isHovered && props.$checked) {
+      return props.$theme.colors.sliderTrackFillSelectedHover;
+    }
```

Comment:
Should there be another condition for just `props.$checked` case (when not active or hovered)?

Reply:
updated the colors on this element. checked/default have the same color, so no need to create another if statement